pkg/operator/controller/dns_status.go

```diff
@@ -114,13 +108,16 @@ func computeDNSDegradedCondition(oldCondition *operatorv1.OperatorCondition, clu
 			status = operatorv1.ConditionTrue
 			degradedReasons = append(degradedReasons, "NoDNSPodsAvailable")
 			messages = append(messages, "No DNS pods are available.")
-		case intstrErr != nil:
-			degradedReasons = append(degradedReasons, "InvalidDNSMaxUnavailable")
-			messages = append(messages, fmt.Sprintf("The DNS daemonset has an invalid MaxUnavailable value: %v", intstrErr))
-		case int(numberUnavailable) > maxUnavailable:
-			status = operatorv1.ConditionTrue
-			degradedReasons = append(degradedReasons, "MaxUnavailableDNSPodsExceeded")
-			messages = append(messages, fmt.Sprintf("Too many DNS pods are unavailable (%d > %d max unavailable).", numberUnavailable, maxUnavailable))
+		case want > have:
+			// Mark as degraded only if it's been determined that it's not already degraded as a recent last transition.
+			if oldCondition != nil && oldCondition.Status == operatorv1.ConditionFalse {
+				timeSinceTransition := currentTime.Sub(oldCondition.LastTransitionTime.Time)
+				if timeSinceTransition > transitionUnchangedToleration {
```

Comment:
Couldn't this logic be reduced to use an existing helper:
```suggestion
				if ! lastTransitionTimeIsRecent(currentTime, oldCondition.LastTransitionTime.Time, transitionUnchangedToleration) {
```

Reply:
I considered it but didn't think it worked that lastTransitionTimeIsRecent returns false if elapsed time < 0.  It's definitely an edge case, that would mean that the old condition last transitioned in the future, but there are still time changes that might reflect that once a year. 

I can reconsider if you like, it's really an edge case.

Reply:
That's okay, I see your point, the edge cases may provide undesirable results here.